xonsh/completer.py

```diff
@@ -130,6 +130,13 @@ def _add_dots(self, paths, prefix):
         if prefix == '..':
             paths.add('../')
 
+    def _add_cdpaths(self, paths, prefix):
+        """Completes current prefix using CDPATH"""
+        env = builtins.__xonsh_env__
+        for cdp in env.get("CDPATH", []):
```

Comment:
Yes, I have to think about it, in fact the default bash behaviour is strange:

    [naufraghi@odello ~]$ cd src/
    /home/naufraghi/Documents/src
    [naufraghi@odello src]$ ls
    xonsh
    [naufraghi@odello src]$ ls xonsh/
    add-CDPATH.diff  CONTRIBUTING  docs  license  README.rst  recipe  requirements.txt  scripts  setup.py  tests  xonsh
    [naufraghi@odello src]$ export CDPATH=$HOME/Documents/src/
    [naufraghi@odello src]$ cd xonsh/
    /home/naufraghi/Documents/src/xonsh
    [naufraghi@odello xonsh]$ cd xonsh/
    /home/naufraghi/Documents/src/xonsh
    [naufraghi@odello xonsh]$ ls
    add-CDPATH.diff  CONTRIBUTING  docs  license  README.rst  recipe  requirements.txt  scripts  setup.py  tests  xonsh
    [naufraghi@odello xonsh]$ cd ./xonsh/
    [naufraghi@odello xonsh]$ ls
    aliases.py     built_ins.py  environ.py   __init__.py    lazyjson.py  openpy.py  prompt_toolkit_completer.py     prompt_toolkit_shell.py  shell.py
    ast.py         completer.py  execer.py    inspectors.py  lexer.py     parser.py  prompt_toolkit_history.py       pyghooks.py              timings.py
    base_shell.py  dirstack.py   imphooks.py  jobs.py        main.py      proc.py    prompt_toolkit_key_bindings.py  readline_shell.py        tools.py